pandas/core/arrays/datetimelike.py

```diff
@@ -464,7 +468,10 @@ def _addsub_offset_array(self, other, op):
                       "{cls} not vectorized"
                       .format(cls=type(self).__name__), PerformanceWarning)
 
-        res_values = op(self.astype('O').values, np.array(other))
```

Comment:
is it worth a branch here in is_extension_dtype?

Reply:
I think `values_from_object` is a pretty clean solution to this.